### PR TITLE
Refs #12990 -- Added example to JSONField release notes.

### PR DESCRIPTION
**Pull request by orf**
As  discussed on IRC, I think the JSONField release notes deserve a code snippet of some kind.

I struggled to think of a suitable, compact example.

### CODE REVIEW

File: docs/releases/3.1.txt [hunk 1]
```diff
@@ -70,8 +70,18 @@ JSONField for all supported database backends
...
+PostgreSQL-only::
+
+    from django.db import models
+
+    class ContactInfo(models.Model):
+        data = models.JSONField()
+
+    ContactInfo.objects.create(data={"name": "John", "city": "London"})
+    ContactInfo.objects.filter(data__name="John") \
+        .update(data__city="Cambridge")
```
Comment (laymonage) [inline docs/releases/3.1.txt:84]: This is not supported until [#29112](https://code.djangoproject.com/ticket/29112) is fixed.

File: docs/releases/3.1.txt [hunk 2]
```diff
@@ -70,8 +70,18 @@ JSONField for all supported database backends
 Django now includes :class:`.models.JSONField` and
 :class:`forms.JSONField <django.forms.JSONField>` that can be used on all
 supported database backends. Both fields support the use of custom JSON
-encoders and decoders. The model field supports the introspection, lookups, and
-transforms that were previously PostgreSQL-only.
+encoders and decoders. The model field supports the :ref:`introspection,
+lookups, and transforms <querying-jsonfield>` that were previously
+PostgreSQL-only::
```
Comment (felixxm) [inline docs/releases/3.1.txt:75]: I would add `querying-jsonfield` reference with an example.

Comment (orf) [reply docs/releases/3.1.txt:75]: Can you elaborate? Sorry, I'm not sure what you mean here. I have a reference to `querying-jsonfield` already?

I wanted to create a small, condensed example snippet that showed off a couple of the querying features - I went for a basic equality query combined with a `__contains` as these seem interesting.

Comment (felixxm) [reply docs/releases/3.1.txt:75]: Sorry, typo it should be _"..reference **without** an example"_. But OK, let's leave an example. I added `has_key`.

Comment (orf) [reply docs/releases/3.1.txt:75]: Awesome, it looks a lot better. Thanks!

I think code samples are like pictures to developers - they tell a thousand words (and our eyes are drawn to them), and we should definitely include them for the big-ticket features where we can 💪

**Pull request merged**